packages/server/src/middleware/nuxt.js

```diff
@@ -125,30 +125,25 @@ const getCspString = ({ cspScriptSrcHashSet, allowedSources, policies, isDev })
   const policyObjectAvailable = typeof policies === 'object' && policies !== null && !Array.isArray(policies)
 
   if (policyObjectAvailable) {
-    const transformedPolicyObject = transformPolicyObject(policies, cspScriptSrcHashSet)
+    const transformedPolicyObject = transformPolicyObject(policies, cspScriptSrcHashes)
 
     return Object.entries(transformedPolicyObject).map(([k, v]) => `${k} ${v.join(' ')}`).join('; ')
   }
 
   return baseCspStr
 }
 
-const transformPolicyObject = (policies, cspScriptSrcHashSet) => {
+const transformPolicyObject = (policies, cspScriptSrcHashes) => {
   const userHasDefinedScriptSrc = policies['script-src'] && Array.isArray(policies['script-src'])
 
   // Self is always needed for inline-scripts, so add it, no matter if the user specified script-src himself.
 
-  const hashAndPolicySet = cspScriptSrcHashSet
-  hashAndPolicySet.add(`'self'`)
+  let hashAndPolicyList = cspScriptSrcHashes
```

Comment:
Do we need the re-assignment here?

Reply:
rewrote this part a bit to avoid modifying input param, please look at it again